File: 347-top-k-frequent-elements/347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
            count = {}
            freq = [[] for i in range(len(nums) + 1)]

            for i in nums :
                count[i] = count.get(i , 0) + 1

            for i , j in count.items():
                freq[j].append(i)
            ans = []
            for i in range(len(freq)-1 ,-1 , -1):
                for j in freq[i]:
                    ans.append(j)
                    if len(ans) == k :
                        return ans

File: 347-top-k-frequent-elements/347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        
	 	 	 int n = nums.size();
	        unordered_map<int, int> m;
	        for (int i = 0; i < n; i++) {
	            m[nums[i]]++;
	        }

	        vector<vector<int>> buckets(n + 1);
	        for (auto it = m.begin(); it != m.end(); it++) {
	            buckets[it->second].push_back(it->first);
	        }

	        vector<int> result;

	        for(int i = n ; i >=0 ; i--){
	        	if((int) result.size() == k)
	        		break;
	        	for(auto j : buckets[i]){
	        		result.push_back(j);
	        	}
	        }
	        return result;
    }
};